### DISCUSSION TITLE
Chat / RAG with bookmarks

### DISCUSSION THREAD

**Discussion by jbohnslav** [karakeep-app/karakeep#709] [Ideas]
Hi, I'm interested in this app or something like it.

I want to be able to chat with my bookmarks. Like, if I downloaded a bunch of blogs about self hosting, I want to use a chat interface to ask "how do I configure cloudflare tunnels" and use RAG over my bookmarks to answer.

Are there plans for a chat box interface / vector DB?

**Comment by MohamedBassem** [comment]
Hey, this is indeed planned. We have work in progress (https://github.com/hoarder-app/hoarder/pull/403) to generate embeddings for all the content you store in hoarder and we're planning to use for RAG applications and semantic search.

**Reply by jbohnslav** [reply]
Awesome, appreciate the link!

**Comment by ovizii** [comment]
I'd also love that. A search where I could ask what was the project which I found on github that does X, Y and Z would be so much easier to find than simply searching for the keywords. I guess that is not quite RAG but an AI enhanced search would be amazing.

**Comment by reedcwilson** [comment]
Any update on this? I noticed that #403 was merged, but that doesn't store the embeddings. I am super excited about a feature like this, so just wanting to check in.

**Comment by louis-pre** [comment]
Bumping here as well, I think Karakeep already makes pretty good use of AI but that would bring it to the next level in my book. 

I noticed a PR exists to move this forward https://github.com/karakeep-app/karakeep/pull/834 but it seems to have stalled a few months back. 

Is there any way the community can help move this forward?

**Comment by Tokynet** [comment]
BUMP

**Comment by dxnxz** [comment]
bump :)

**Comment by mihaime** [comment]
bump, was this forgotten?

**Comment by MohamedBassem** [comment]
embedding generation was merged two weeks ago and is used for tag generation. Chat is coming in the next release, it's being polished.

**Reply by louis-pre** [reply]
Nice! Is there a plan to also integrate it into the search?
